react/features/base/ui/middleware.web.js

```diff
@@ -0,0 +1,39 @@
+// @flow
+
+import { i18next } from '../i18n';
+import { I18NEXT_INITIALIZED, LANGUAGE_CHANGED } from '../i18n/actionTypes';
+import { MiddlewareRegistry } from '../redux';
+
+import { setDirection } from './actions';
+import { detectDirectionBasedOnLanguage } from './functions';
+
+/**
+ * Middleware that captures I18NEXT_INITIALIZED and LANGUAGE_CHANGED actions and
+ * change the direction of the app based on the language that is set.
+ *
+ * @param {Store} store - The redux store.
+ * @returns {Function}
+ */
+MiddlewareRegistry.register(store => next => action => {
+    switch (action.type) {
+    case I18NEXT_INITIALIZED:
+    case LANGUAGE_CHANGED :{
+
+        // XXX Right now, removing the following line will mess up with the app layout.
```

Comment:
I think this can go now that the rest of the PRs have landed now, right?

Reply:
No. There are still changes that need to be made. If you select a RTL language, and u open chat, it will break the layout. Or Filmstrips will break.